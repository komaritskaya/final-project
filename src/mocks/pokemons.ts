import { Pokemon } from '../types';

const pokemons: Pokemon[] = [
  {
    name: 'bulbasaur',
    id: 1,
    isCaught: true,
    catchDate: new Date(),
  },
  {
    name: 'ivysaur',
    id: 2,
    isCaught: false,
    catchDate: null,
  },
  {
    name: 'venusaur',
    id: 3,
    isCaught: false,
    catchDate: null,
  },
  {
    name: 'charmander',
    id: 4,
    isCaught: false,
    catchDate: null,
  },
];

export default pokemons;
