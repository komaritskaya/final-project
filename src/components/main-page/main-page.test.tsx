import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import renderer from 'react-test-renderer';
import MainPage from './main-page';
import pokemons from '../../mocks/pokemons';

const mockStore = configureStore([]);

it('Should MainPage component render correctly', () => {
  const store = mockStore({
    data: {
      pokemons,
    },
    app: {
      shownPokemonsCount: 1,
    },
  });

  const tree = renderer
    .create(
      <Provider store={store}>
        <Router>
          <MainPage />
        </Router>
      </Provider>,
    ).toJSON();

  expect(tree).toMatchSnapshot();
});
