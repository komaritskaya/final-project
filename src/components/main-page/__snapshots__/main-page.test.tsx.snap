// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should MainPage component render correctly 1`] = `
Array [
  <ul>
    <li>
      <div
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex={0}
      >
        default
      </div>
    </li>
    <li>
      <div
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex={0}
      >
        caught
      </div>
    </li>
  </ul>,
  <div>
    <div>
      <a
        href="/1"
        onClick={[Function]}
      >
        bulbasaur
      </a>
    </div>
    <div>
      <a
        href="/2"
        onClick={[Function]}
      >
        ivysaur
      </a>
    </div>
    <div>
      <a
        href="/3"
        onClick={[Function]}
      >
        venusaur
      </a>
    </div>
    <div>
      <a
        href="/4"
        onClick={[Function]}
      >
        charmander
      </a>
    </div>
    <div>
      <a
        href="/5"
        onClick={[Function]}
      >
        charmeleon
      </a>
    </div>
    <div>
      <a
        href="/6"
        onClick={[Function]}
      >
        charizard
      </a>
    </div>
    <div>
      <a
        href="/7"
        onClick={[Function]}
      >
        squirtle
      </a>
    </div>
    <div>
      <a
        href="/8"
        onClick={[Function]}
      >
        wartortle
      </a>
    </div>
    <div>
      <a
        href="/9"
        onClick={[Function]}
      >
        blastoise
      </a>
    </div>
    <div>
      <a
        href="/10"
        onClick={[Function]}
      >
        caterpie
      </a>
    </div>
    <div>
      <a
        href="/11"
        onClick={[Function]}
      >
        metapod
      </a>
    </div>
    <div>
      <a
        href="/12"
        onClick={[Function]}
      >
        butterfree
      </a>
    </div>
    <div>
      <a
        href="/13"
        onClick={[Function]}
      >
        weedle
      </a>
    </div>
    <div>
      <a
        href="/14"
        onClick={[Function]}
      >
        kakuna
      </a>
    </div>
    <div>
      <a
        href="/15"
        onClick={[Function]}
      >
        beedrill
      </a>
    </div>
    <div>
      <a
        href="/16"
        onClick={[Function]}
      >
        pidgey
      </a>
    </div>
    <div>
      <a
        href="/17"
        onClick={[Function]}
      >
        pidgeotto
      </a>
    </div>
    <div>
      <a
        href="/18"
        onClick={[Function]}
      >
        pidgeot
      </a>
    </div>
    <div>
      <a
        href="/19"
        onClick={[Function]}
      >
        rattata
      </a>
    </div>
    <div>
      <a
        href="/20"
        onClick={[Function]}
      >
        raticate
      </a>
    </div>
  </div>,
  <div>
    <button
      onClick={[Function]}
      type="button"
    >
      Show more
    </button>
  </div>,
]
`;
